<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="grad_field_local_planner/GradFieldPlannerROS" />
    <rosparam file="$(find grad_field_local_planner)/params/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find grad_field_local_planner)/params/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find grad_field_local_planner)/params/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find grad_field_local_planner)/params/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find grad_field_local_planner)/params/move_base_params.yaml" command="load" />
    <rosparam file="$(find grad_field_local_planner)/params/gf_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
</launch>
